flows/engine/engine.go

```diff
@@ -143,24 +139,30 @@ func resumeNode(run flows.FlowRun, node flows.Node, step flows.Step, event flows
 		return noDestination, step, nil
 	}
 
-	err := wait.End(run, step, event)
+	// try to resume our wait with the first caller event
+	err := wait.End(run, step, callerEvents[0])
```

Comment:
If order is important then the wait is waiting for the _next_ event, i.e. the first in the list. This feels right IMO